src/librbd/watcher/Notifier.h

```diff
@@ -25,21 +27,21 @@ class Notifier {
   ~Notifier();
 
   void flush(Context *on_finish);
-  void notify(bufferlist &bl, bufferlist *out_bl, Context *on_finish);
+  void notify(bufferlist &bl, NotifyResponse *response, Context *on_finish);
 
 private:
   typedef std::list<Context*> Contexts;
 
   struct C_AioNotify : public Context {
     Notifier *notifier;
+    NotifyResponse *response;
     Context *on_finish;
+    bufferlist *out_bl;
```

Comment:
Nit: any reason why we can't just inline the bufferlist instead of conditionally allocating the buffer if a ```NotifyResponse``` was provided? This isn't a high-profile path so for the cleaner code, I think we can handle the potentially wasted memory allocation.